cmd/process-agent/main_common.go

```diff
@@ -111,6 +112,9 @@ func runApp(ctx context.Context, globalParams *command.GlobalParams) error {
 		Config config.Component
 	}
 	app := fx.New(
+		// Increase default fx start/stop timeout to account for delays caused by system load.
+		fx.StartTimeout(5*time.Minute),
```

Comment:
Feels like a large number for a timeout, to be frank. Have we experimented with other values?

Reply:
discussed on slack, summarizing here

Yes it is a high value, but we didn’t any any timeout at all before fx. So any value is more restrictive than it was before.
If we choose a value too high, it hangs for that period of time, where before fx it would have hung forever. If we choose a value too small, then the agent won't start and could impact customers. So let's choose a large value for now to reduce chances of impacting customers while we migrate to fx.